profile.go

```diff
@@ -0,0 +1,38 @@
+package slack
+
+import (
+	"context"
+	"encoding/json"
+	"net/url"
+)
+
+// SetUserProfile will set a profile for the provided user
+func (api *Client) SetUserProfile(user string, profile *UserProfile) error {
+	return api.SetUserProfileContext(context.Background(), user, profile)
+}
+
+// SetUserProfileContext will set a profile for the provided user with a custom context
+func (api *Client) SetUserProfileContext(ctx context.Context, user string, profile *UserProfile) error {
+	jsonProfile, err := json.Marshal(profile)
```

Comment:
Each field in UserProfile is not omitempty, so when it is converted to json, it will all have initial values. So it may occur unintended updates.

Reply:
How do you think we can correct this behavior?